espresso/streamer.go

```diff
@@ -251,71 +334,51 @@ func (s *EspressoStreamer[B]) Update(ctx context.Context) error {
 		s.Log.Trace("Remaining list", "Inserting batch into buffer", "batch", batch)
 		s.BatchBuffer.Insert(batch, pos)
 		delete(s.RemainingBatches, k)
-
 	}
+}
 
-	s.Log.Info("Remaining list after", "Size", len(s.RemainingBatches))
-
-	// Process the new batches fetched from Espresso
-	for ; i <= finish; i++ {
-		s.Log.Trace("Fetching HotShot block", "block", i)
-
-		txns, err := s.EspressoClient.FetchTransactionsInBlock(ctx, i, s.Namespace)
+// processEspressoTransactions is a helper method that encapsulates the logic of
+// processing batches from the transactions in a block fetched from Espresso.
+func (s *EspressoStreamer[B]) processEspressoTransactions(ctx context.Context, i uint64, txns espressoClient.TransactionsInBlock) {
+	for _, transaction := range txns.Transactions {
+		batch, err := s.UnmarshalBatch(transaction)
 		if err != nil {
-			return fmt.Errorf("Failed to fetch transactions in block: %w", err)
-		}
-
-		s.Log.Trace("Fetched HotShot block", "block", i, "txns", len(txns.Transactions))
-
-		if len(txns.Transactions) == 0 {
-			s.Log.Trace("No transactions in hotshot block", "block", i)
+			s.Log.Warn("Dropping batch with invalid transaction data", "error", err)
 			continue
 		}
 
-		for _, transaction := range txns.Transactions {
+		s.Log.Info("Inserting batch into buffer", "batch", batch)
```

Comment:
Nit: we aren't yet inserting here, in fact we have exact same message on the penultimate line of the function, which could be confusing when grepping through logs.